Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp

```diff
@@ -395,82 +397,117 @@ String WebsiteDataStore::migrateMediaKeysStorageIfNecessary(const String& direct
+
+    if (!directories.cacheStorageDirectory.isEmpty())
+        directories.cacheStorageDirectory = resolveAndCreateReadWriteDirectoryForSandboxExtension(directories.cacheStorageDirectory);
+
+    if (!directories.hstsStorageDirectory.isEmpty())
+        directories.hstsStorageDirectory = resolveAndCreateReadWriteDirectoryForSandboxExtension(directories.hstsStorageDirectory);
+
+    if (!directories.generalStorageDirectory.isEmpty())
+        directories.generalStorageDirectory = resolveAndCreateReadWriteDirectoryForSandboxExtension(directories.generalStorageDirectory);
+
+    if (!directories.searchFieldHistoryDirectory.isEmpty())
+        directories.searchFieldHistoryDirectory = resolveAndCreateReadWriteDirectoryForSandboxExtension(directories.searchFieldHistoryDirectory);
+
+    if (!directories.generalStorageDirectory.isEmpty())
+        directories.generalStorageDirectory = resolveAndCreateReadWriteDirectoryForSandboxExtension(directories.generalStorageDirectory);
+
 #if ENABLE(ARKIT_INLINE_PREVIEW)
-    if (!configuration->modelElementCacheDirectory().isEmpty())
-        resolvedConfiguration->setModelElementCacheDirectory(resolveAndCreateReadWriteDirectoryForSandboxExtension(configuration->modelElementCacheDirectory()));
+    if (!directories.modelElementCacheDirectory.isEmpty())
+        directories.modelElementCacheDirectory = resolveAndCreateReadWriteDirectoryForSandboxExtension(directories.modelElementCacheDirectory);
 #endif
-    if (!configuration->searchFieldHistoryDirectory().isEmpty())
-        resolvedConfiguration->setSearchFieldHistoryDirectory(resolveAndCreateReadWriteDirectoryForSandboxExtension(configuration->searchFieldHistoryDirectory()));
 
-    // Resolve file paths.
-    if (!configuration->cookieStorageFile().isEmpty()) {
-        auto resolvedCookieDirectory = resolveAndCreateReadWriteDirectoryForSandboxExtension(FileSystem::parentPath(configuration->cookieStorageFile()));
-        resolvedConfiguration->setCookieStorageFile(FileSystem::pathByAppendingComponent(resolvedCookieDirectory, FileSystem::pathFileName(configuration->cookieStorageFile())));
+    if (!directories.cookieStorageFile.isEmpty()) {
+        auto resolvedCookieDirectory = resolveAndCreateReadWriteDirectoryForSandboxExtension(FileSystem::parentPath(directories.cookieStorageFile));
+        directories.cookieStorageFile = FileSystem::pathByAppendingComponent(resolvedCookieDirectory, FileSystem::pathFileName(directories.cookieStorageFile));
     }
+}
 
-    // Default paths of WebsiteDataStore created with identifier are not under caches or tmp directory,
-    // so we need to explicitly exclude them from backup.
-    if (configuration->identifier()) {
-        Vector<String> allCacheDirectories = {
-            resolvedMediaCacheDirectory()
-            , resolvedNetworkCacheDirectory()
+WebsiteDataStoreConfiguration::Directories WebsiteDataStore::resolvedDirectories()
```

Comment:
Given that `m_resolvedDirectories` MUST have been set by the bg thread before this function has been called, as given that the bg thread can never overwrite the data member later on, I believe it is safe to return a `WebsiteDataStoreConfiguration::Directories`. Also, the member function should be marked as `const`.

Finally, every call site can store in an `auto&` instead of `auto`.

Reply:
I think the reason it cannot be marked as const is the lock; I will try making the change and check.